step-02/js/main.js

```diff
@@ -1,239 +1,278 @@
+        `videoHeight: ${video.videoHeight}px.`);
+};
+
+// Logs a message with the id and size of a video element.
+const logResizedVideo = (event) => {
+  logVideoLoaded(event);  // Log size of video by calling logVideoLoaded.
+
+  // First on-resize is used as an indication that video started playing out.
+  if (startTime) {  // If startTime is defined is first load.
+    const elapsedTime = window.performance.now() - startTime;
     startTime = null;
+    trace(`Setup time: ${elapsedTime.toFixed(3)}ms.`);
+  }
+};
+
+localVideo.addEventListener('loadedmetadata', logVideoLoaded);
+remoteVideo.addEventListener('loadedmetadata', logVideoLoaded);
+remoteVideo.addEventListener('onresize', logResizedVideo);
+
+
+// Define RTC peer connection behavior.
+
+// Connects with new peer candidate.
+const connectionHandler = (event) => {
+  const peerConnection = event.target;
+  const iceCandidate = event.candidate;
+
+  if (iceCandidate) {
+    const newIceCandidate = new RTCIceCandidate(iceCandidate);
+    const otherPeer = getOtherPeer(peerConnection);
+
+    otherPeer.addIceCandidate(newIceCandidate)
+      .then(() => { connectionSucceeded(peerConnection); })
+      .catch((error) => { connectionFailed(peerConnection, error); });
+
+    trace(`${getPeerName(peerConnection)} ICE candidate:\n` +
+          `${event.candidate.candidate}.`);
   }
 };
 
-var localStream;
-var pc1;
-var pc2;
-var offerOptions = {
-  offerToReceiveAudio: 1,
-  offerToReceiveVideo: 1
+// Logs that the connection succeeded.
+const connectionSucceeded = (peerConnection) => {
+  trace(`${getPeerName(peerConnection)} addIceCandidate success.`);
 };
 
-function getName(pc) {
-  return (pc === pc1) ? 'pc1' : 'pc2';
+// Logs that the connection failed.
+const connectionFailed = (peerConnection, error) => {
+  trace(`${getPeerName(peerConnection)} failed to add ICE Candidate:\n`+
+        `${error.toString()}.`);
 }
 
-function getOtherPc(pc) {
-  return (pc === pc1) ? pc2 : pc1;
+// Logs changes to the connection state.
+const connectionChanged = (event) => {
+  const peerConnection = event.target;
+  console.log('ICE state change event: ', event);
+  trace(`${getPeerName(peerConnection)} ICE state: ` +
+        `${peerConnection.iceConnectionState}.`);
+};
+
+// Logs error when setting session description fails.
+const setSessionDescriptionError = (error) => {
+  trace(`Failed to create session description: ${error.toString()}.`);
+};
+
+// Logs success when setting session description.
+const setDescriptionSuccess = (peerConnection, functionName) => {
+  const peerName = getPeerName(peerConnection);
+  trace(`${peerName} ${functionName} complete.`);
 }
 
-function gotStream(stream) {
-  trace('Received local stream');
-  localVideo.srcObject = stream;
-  // Add localStream to global scope so it's accessible from the browser console
-  window.localStream = localStream = stream;
-  callButton.disabled = false;
+// Short-hand for setDescriptionSuccess when using setLocalDescription.
```

Comment:
I don't think 'Short-hand' is clear here.

Maybe 'Called when the localDescription is set successfully ' (or similar).

Reply:
Done.